core/src/main/java/com/orientechnologies/orient/core/db/record/ORecordLazyList.java

```diff
@@ -154,20 +154,23 @@ public boolean contains(final Object o) {
 
   @Override
   public boolean add(OIdentifiable e) {
-    if (e != null) {
-      if ((ridOnly || contentType == MULTIVALUE_CONTENT_TYPE.ALL_RIDS)
-          && e.getIdentity().isPersistent()
-          && (e instanceof ODocument && !((ODocument) e).isDirty()))
-        // IT'S BETTER TO LEAVE ALL RIDS AND EXTRACT ONLY THIS ONE
-        e = e.getIdentity();
-      else contentType = ORecordMultiValueHelper.updateContentType(contentType, e);
-    }
-    lazyLoad(true);
+    preAdd(e);
     return super.add(e);
   }
 
   @Override
   public void add(int index, OIdentifiable e) {
+    preAdd(e);
+    super.add(index, e);
+  }
+  
+  @Override
+  public boolean addInternal(OIdentifiable e) {
+	preAdd(e);
```

Comment:
in this case the lazy load should not run, so is not correct to use preAdd as it is here.

Reply:
@tglman , that's actually the actual fix. Without it - my testcase will fail.

Reply:
Without calling `preAdd(a)` - list will not update `contentType` and first iterator will switch `contentType` from EMPTY to ALL_RECORDS which prevents the conversion of the rest of the items in the list.

Reply:
Next commit is fixing this